packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualRows.ts

```diff
@@ -233,10 +233,14 @@ export const useGridVirtualRows = (apiRef: GridApiRef): void => {
         const currentRowPage = elementIndex / gridState.containerSizes!.viewportPageSize;
         const viewportHeight = gridState.containerSizes!.renderingZoneScrollHeight;
         const scrollPosition = currentRowPage * viewportHeight;
-        const pageHeight = gridState.containerSizes!.viewportPageSize * rowHeight;
+
+        // The amount of space taken by the first partially visible row
+        const partialRowHeight = windowRef.current!.scrollTop % rowHeight;
 
         isRowIndexAbove = windowRef.current!.scrollTop > scrollPosition;
-        isRowIndexBelow = scrollPosition + rowHeight > windowRef.current!.scrollTop + pageHeight;
+        isRowIndexBelow =
+          scrollPosition + rowHeight >
+          windowRef.current!.scrollTop + viewportHeight + partialRowHeight;
```

Comment:
Fixes a bug where the grid wasn't scrolling if the row was partially visible.

<img width="500" src="https://user-images.githubusercontent.com/42154031/121949061-6f094400-cd2e-11eb-9815-5601d2b1f170.gif" alt="BPYpligb7o" style="max-width:100%;">